HW_8/task 8.2.js

```diff
@@ -0,0 +1,269 @@
+
+// Пример:
+// addDepartment(1, "Название нового отдела")
+
+// Функция-помощник для получения предприятия по id предприятия
+const getEnterpriseById = (id, []) =>
+enterprises.filter((enterprise) => enterprise.id === id);
+
+const addDepartment = (id, newDepartment) => {
+  const enterpriseFound = getEnterpriseById(id, enterprises);
+  if (enterpriseFound != 0) {
+    enterpriseFound[0].departments.push({
+      id: newId(enterprises),
+      name: newDepartment,
+      employees_count: null,
+    });
+    
+    return JSON.stringify(enterprises);
+  }
+  
+  throw new Error('Предприятия с данным id не существует');
+}
+
+console.log(addDepartment(1, "Отдел бездельников"));
+
+// 5. Написать функцию для редактирования названия предприятия. Принимает в качестве аргумента id предприятия и новое имя предприятия.
+
+// Пример:
+// editEnterprise(1, "Новое название предприятия")
+
+const editEnterprise = (id, newName) => {
+  const foundEnterprise = getEnterpriseById(id, enterprises);
+  if (foundEnterprise == 0) throw new Error('Предприятия с данным id не существует');
+  foundEnterprise[0].name = newName;
+  
+  return JSON.stringify(enterprises);
+}
+
+console.log(editEnterprise(1, 'ООО "Рога и копыта"'));
+
+// 6. Написать функцию для редактирования названия отдела. Принимает в качестве аргумента id отдела и новое имя отдела.
+
+// Пример:
+// editDepartment(7, "Новое название отдела")
+
+// Функция-помощник для получения отдела по id отдела
+const getDepartmentById = (id, []) => {
+  let result;
+  enterprises.forEach((enterprise) => enterprise.departments.forEach((department) => {
+      if (department.id === id) {
+        result = department;
+      }
+    })
+  );
+  
+  return result;
+};
+
+const editDepartment = (id, newName) => {
+  const foundDepartment = getDepartmentById(id, enterprises);
+  if (foundDepartment == undefined) throw new Error('Отдела с данным id не существует');
+  foundDepartment.name = newName;
+  
+  return JSON.stringify(enterprises);
+}
+
+console.log(editDepartment(7, "Новое название отдела"));
+
+// 7. Написать функцию для удаления предприятия. В качестве аргумента принимает id предприятия.
+
+// Пример:
+// deleteEnterprise(1)
+
+const deleteEnterprise = (id) => {
+  const foundEnterprise = getEnterpriseById(id, enterprises);
+  if (foundEnterprise.length === 0) throw new Error('Предприятия с данным id не существует');
+  
+  const findIndex = enterprises.indexOf(...foundEnterprise);
+  enterprises.splice(findIndex, 1)
+  
+  return enterprises;
+}
+
+console.log(deleteEnterprise(1));
+
+// 8. Написать функцию для удаления отдела. В качестве аргумента принимает id отдела. Удалить отдел можно только, если в нем нет сотрудников.
+
+// Пример:
+// deleteDepartment(3)
+
+const deleteDepartment = (id) => {
+  if (typeof id != 'number') throw new Error('Введите верные данные');
+  const foundEnterprise = getEnterpriseByDepartment(id);
```

Comment:
```suggestion
  const foundEnterprise = getEnterpriseByDepartment(id); // as option u can return the enterprise itself, not an array
```

Reply:
The code with your suggestion looks cleaner, I'll add it to the new commit